### PR TITLE
private/marco/anim engine trans filter for shapes

### PR DESCRIPTION
**Pull request by mcecchetti**
- **slideshow: refactoring: move from SlideInfo to TransitionFilterInfo**
- **slideshow: anim engine: displaySlide, check if we're still presenting**
- **slideshow: initial support for transition filters for shapes**

### CODE REVIEW

File: browser/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "@types/jquery.contextmenu": "^1.7.38",
     "@types/jqueryui": "^1.12.21",
     "@types/mocha": "8.2.0",
-    "@types/node": "14.14.25",
+    "@types/node": "18.11.09",
     "@types/offscreencanvas": "^2019.7.3",
```
Comment (eszkadev) [inline browser/package.json:12]: Please do not change that if possibleour CI as we experienced in the past. Why is that needed?

File: browser/src/slideshow/engine/AnimationFactory.ts [hunk 2]
```diff
@@ -347,16 +519,22 @@ function createShapeTransition(
...
+				transitionFilterInfo,
+				aAnimatedElement,
 			);
+
+			// const aParametricPolyPolygon = createClipPolyPolygon(
+			// 	eTransitionType,
+			// 	eTransitionSubType,
+			// );
+			// const aClippingAnimation = new ClippingAnimation(
+			// 	aParametricPolyPolygon,
```
Comment (eszkadev) [inline browser/src/slideshow/engine/AnimationFactory.ts:533]: dead comment? or for debbuging later?

File: browser/package.json (hunk 1 above)
Comment (mcecchetti) [reply browser/package.json:12]: I upgraded version for structuredClone() function looks nice for cloning arrays, maps.

File: browser/src/slideshow/engine/AnimationFactory.ts (hunk 2 above)
Comment (mcecchetti) [reply browser/src/slideshow/engine/AnimationFactory.ts:533]: I'll remove when work on transition filter is over.

**Review by eszkadev — COMMENTED** [review]
it looks good, but please bring back   NODE version :)

File: browser/package.json (hunk 1 above)
Comment (eszkadev) [reply browser/package.json:12]: ah it's just types :) it'as fine then

**Review by eszkadev — APPROVED** [review]

**Pull request merged**